### PR TITLE
[JSDoc] Add missing documentation for botbuilder-ai-orchestrator

### PR DESCRIPTION
**Pull request by sw-joelmut**
Fixes # 2602

## Description
This PR adds all the missing JSDoc documentation for [botbuilder-ai-orchestrator](https://github.com/microsoft/botbuilder-js/tree/main/libraries/botbuilder-ai-orchestrator) based on the errors thrown. Some documentation is ported and adapted from [botbuilder-dotnet](https://github.com/microsoft/botbuilder-dotnet).

## Specific Changes

- Ported documentation from botbuilder-dotnet
- Added `@private` for private methods.
- Added JSDoc comments in **3 files**.

## Testing
In the following image there is a sneak peek about the ESLint Report with no JSDoc errors.
image

### CODE REVIEW

File: libraries/botbuilder-ai-orchestrator/src/orchestratorComponentRegistration.ts [hunk 1]
```diff
@@ -12,14 +12,25 @@ import { BuilderRegistration, ComponentRegistration, ResourceExplorer } from 'bo
...
 
+/**
+ * Define component assets for Luis.
+ */
 export class OrchestratorComponentRegistration implements ComponentRegistration {
     private readonly _builderRegistrations: BuilderRegistration[] = [];
     private _resourceExplorer: ResourceExplorer;
 
+    /**
+     * Gets all the builder registrations instances.
```
Comment (santgr11) [inline libraries/botbuilder-ai-orchestrator/src/orchestratorComponentRegistration.ts:23]: I think the plural here goes only to instances, but I'm not sure.
```suggestion
     * Gets all the builder registration instances.
```

**Review by santgr11 — APPROVED** [review]

File: libraries/botbuilder-ai-orchestrator/src/orchestratorComponentRegistration.ts (hunk 1 above)
Comment (sw-joelmut) [reply libraries/botbuilder-ai-orchestrator/src/orchestratorComponentRegistration.ts:23]: I think `registrations` its ok.
Maybe @ceciliaavila or @iscaltritti-sw can apport on this.

Comment (iscaltritti-sw) [reply libraries/botbuilder-ai-orchestrator/src/orchestratorComponentRegistration.ts:23]: I agree with @santgr11, hard to explain grammatically. 
Registrations instances would be correct for a matrix where each builder gets more than one registration.

**Review by iscaltritti-sw — CHANGES_REQUESTED** [review]

**Review by ceciliaavila — APPROVED** [review]
With the fix that santgr11 mentioned, LGTM.

**Comment by sw-joelmut** [comment]
@santgr11 @ceciliaavila @iscaltritti-sw, applied feedback, thanks

**Review by iscaltritti-sw — APPROVED** [review]

**Review by iscaltritti-sw — APPROVED** [review]

**Review by denscollo — APPROVED** [review]

**Review by ceciliaavila — APPROVED** [review]

**Comment by denscollo** [comment]
promoted

**Pull request closed without merging**